routing/payment_lifecycle.go

```diff
@@ -1108,3 +1076,100 @@ func (p *paymentLifecycle) processResultsAndReloadPayment() (DBMPPayment,
 
 	return payment, ps, nil
 }
+
+// processSwitchResults reads the `p.results` map and process the results
+// returned from the htlcswitch.
+func (p *paymentLifecycle) processSwitchResults() error {
+	// Create a slice to remember the results of the attempts that we have
+	// processed.
+	attempts := make([]*channeldb.HTLCAttempt, 0, p.switchResults.Len())
+
+	var errReturned error
+
+	// Range over the map to process all the results.
+	p.switchResults.Range(func(a *channeldb.HTLCAttempt,
```

Comment:
Ok, looks like it's only ever used along with map iteration. In this case, I think we can also just make the key a direct value to ensure we always store a clean copy.

Reply:
I wonder if we should actually use a good ol fashioned mutex here 🤠? `Range` may not get a consistent snapshot of the map, as it's possible that items are added _while_ we're iterating over the map. 

See the implementation here: https://cs.opensource.google/go/go/+/refs/tags/go1.23.6:src/sync/map.go;l=473

Reply:
```
// Range does not necessarily correspond to any consistent snapshot of the Map's
// contents: no key will be visited more than once, but if the value for any key
// is stored or deleted concurrently (including by f), Range may reflect any
// mapping for that key from any point during the Range call. Range does not
// block other methods on the receiver; even f itself may call any method on m.
```

Reply:
TIL the `Range`! I did try to change it to use a simple map plus mutex, then I realized there's an even simpler way to fix it as we can just handle the switch result when we receive it from the `p.resultCollected` chan.